### ISSUE TITLE
Suggest using nameof() for referring to a source for TestCaseSource

### ISSUE THREAD

**Issue by pyrocumulus** [nunit/nunit.analyzers#211]
When referring to a `sourceName` for a `TestCaseSource` it would be nice to suggest using the `nameof()` operation instead of a literal reference. The latter is susceptible to typing mistakes or forgotten renames.

So in short, suggest that:
```csharp
public static IEnumerable MyTestSource
{
    get
    {
        yield return new TestCaseData(1).Returns(true);
        yield return new TestCaseData(0).Returns(false);
    }
}

[TestCaseSource(typeof(MyClass), "MyTestSource")]
public bool Value_ShouldEqual_One(int value)
{
    return value == 1;
}
```

Be altered to:
```csharp
public static IEnumerable MyTestSource
{
    get
    {
        yield return new TestCaseData(1).Returns(true);
        yield return new TestCaseData(0).Returns(false);
    }
}

[TestCaseSource(typeof(MyClass), nameof(MyTestSource))]
public bool Value_ShouldEqual_One(int value)
{
    return value == 1;
}
```

I do not see why it should interfere with other analyzers.

**Comment by mikkelbu** [comment]
We already handle the simple case in NUnit1002 where we don't supply the type name. I'm working on #204 which will also make it work for the case. I hope to have a pre-release of this today or tomorrow

**Comment by pyrocumulus** [comment]
Alright, I'm looking forward to it :)

Op zo 26 apr. 2020 11:47 schreef Mikkel Nylander Bundgaard <
notifications@github.com>:

**Comment by mikkelbu** [comment]
This should be working in the newer builds from myget - https://www.myget.org/feed/nunit-analyzers/package/nuget/NUnit.Analyzers and will be part of the 0.3 release, which I plan to release within a week or two (I just need to squeeze some more functionality into the release 😄).

**Comment by mikkelbu** [comment]
The codefix is now officially released in version 0.3.0 on NuGet - https://www.nuget.org/packages/NUnit.Analyzers, so I'll close this issue. Thanks for reporting it @pyrocumulus 👍.
